src/matter/fabric/FabricManager.ts

```diff
@@ -19,7 +19,10 @@ export class FabricManager {
     }
 
     removeFabric(fabricIndex: FabricIndex) {
-        this.fabrics.splice(this.fabrics.findIndex(fabric => fabric.fabricIndex.index === fabricIndex.index), 1);
+        const index = this.fabrics.findIndex(fabric => fabric.fabricIndex.index === fabricIndex.index);
+        if (!index) return false;
```

Comment:
Instead of return false, it would be better to throw an Error:
* this makes it obvious what is expected code path versus unexpected path
* there is no not obvious boolean return value from the remove method
* this is consistent with the rest of the codebase: Errors are thrown for unexpected parameters / value sand it is up to the caller to handle them, with a catch-all net in the Messenger to send back a basic error

Reply:
ok